client/src/components/core/TeamSelector/TeamSelector.test.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+import "@testing-library/jest-dom/extend-expect";
+import { cleanup, render, fireEvent } from "@testing-library/react";
+import { useTeamSelector } from "./useTeamSelector";
+
+import { useService } from "../../../service";
+import { StoreContext, ActionType } from "../../../store/store";
+
+// Mock the module, without implementing a replacement, this gets hoisted above `mockFetchTeams` so we can't use that here
```

Comment:
Yeah, I think this needs to be documented because I do it just infrequently enough that I forget about this every time, but frequently enough that it's a huge time waster. Maybe I'll document it outside of the code. Basically, in other testing frameworks this would work:

```

const mockFunction = jest.fn();

jest.mock("../../../service", () => ({
  useService: mockFunction,
}));

...
// in the test
expect(mockFunction).toBeCalled();
```

But in jest this doesn't work because of how they implement their mocks, and `mockFunction` won't be defined when the mock is created. Instead you have to first replace the module with a simple mock, then in the test, set the implementation of the mock to something that is defined in the test's scope.